docs/source/using-federation/apollo-gateway-setup.mdx

```diff
@@ -0,0 +1,458 @@
+  },
+  // highlight-end
+});
+
+const server = new ApolloServer({
+  gateway,
+  plugins: [
+    // highlight-start
+    {
+      requestDidStart() {
+        return {
+          async willSendResponse({ contextValue, response }) {
+            // Append our final result to the outgoing response headers
+            response.http.headers.set(
+              'Server-Id',
+              contextValue.serverIds.join(','),
+            );
+          },
+        };
+      },
+    },
+    // highlight-end
+  ],
+});
+
+const { url } = await startStandaloneServer(server, {
+  context() {
+    return { serverIds: [] };
+  },
+});
+
+console.log(`🚀  Server ready at ${url}`);
+```
+
+</ExpansionPanel>
+
+> In this example, multiple calls to `didReceiveResponse` are `push`ing a value onto the shared `contextValue.serverIds` array. The order of these calls cannot be guaranteed. If you write logic that modifies the shared `contextValue` object, make sure that modifications are not destructive, and that the order of modifications doesn't matter.
+
+To learn more about `buildService` and `RemoteGraphQLDataSource`, see the [API docs](./api/apollo-gateway/).
+
+## Custom directive support
+
+The `@apollo/gateway` library supports the use of custom [directives](../schema/directives/) in your subgraph schemas. This support differs depending on whether a given directive is a **type system directive** or an **executable directive**.
+
+### Type system directives
+
+Type system directives are directives that are applied to one of [these locations](http://spec.graphql.org/June2018/#TypeSystemDirectiveLocation). These directives are _not_ used within operations, but rather are applied to locations within the schema itself.
+
+The `@deprecated` directive below is an example of a type system directive:
+
+```graphql
+directive @deprecated(
+  reason: String = "No longer supported"
+) on FIELD_DEFINITION | ENUM_VALUE
+
+type ExampleType {
+  newField: String
+  oldField: String @deprecated(reason: "Use `newField`.")
+}
+```
+
+The gateway strips all definitions _and_ uses of type system directives from your graph's [API schema](/federation/federated-types/overview/#api-schema). This has no effect on your subgraph schemas, which retain this information.
+
+Effectively, the gateway supports type system directives by _ignoring_ them, making them the responsibility of the subgraphs that define them.
+
+### Executable directives
+
+Executable directives are directives that are applied to one of [these locations](http://spec.graphql.org/June2018/#ExecutableDirectiveLocation). These directives are _defined_ in your schema, but they're _used_ in operations that are sent by clients.
+
+> **Although the `@apollo/gateway` library supports executable directives, Apollo Server itself does not**. This guidance is provided primarily for architectures that use the `@apollo/gateway` library in combination with subgraphs that do _not_ use Apollo Server.
```

Comment:
I have no idea what this means. Apollo Server lets you define executable directives in your schema. It doesn't come with any special high-level ways of making your resolver code sensitive to them without writing your own code to manually look at the parsed syntax tree of the operation, but you can still do that if you want!

Reply:
If this note isn't correct does it feel good to you if I just remove it? It doesn't seem like there is anything else we'd need to add here.